cmd/cstor-pool-mgmt/cstorops/pool/pool.go

```diff
@@ -61,37 +88,30 @@ func CreatePool(cStorPoolUpdated *apis.CStorPool) error {
 	}
 
 	createAttr = append(createAttr, cStorPoolUpdated.Spec.PoolSpec.PoolName)
-	if len(cStorPoolUpdated.Spec.Disks.DiskList) < 1 {
-		return fmt.Errorf("Disk name(s) cannot be empty")
-	}
+
 	for _, disk := range cStorPoolUpdated.Spec.Disks.DiskList {
 		createAttr = append(createAttr, disk)
 	}
 
 	//execute pool creation command.
-	poolCreateCmd := exec.Command("zpool", createAttr...)
-	glog.V(4).Info("poolCreateCmd : ", poolCreateCmd)
-	stdoutStderr, err := poolCreateCmd.CombinedOutput()
-	if err != nil {
-		glog.Error("err: ", err)
-		glog.Error("stdoutStderr: ", string(stdoutStderr))
-		return err
-	}
-	glog.Info("Creating Pool Successful")
-	return nil
+	poolCreateCmd := exec.Command(PoolOperator, createAttr...)
+	return poolCreateCmd
 }
 
 // CheckValidPool checks for validity of CStorPool resource.
 func CheckValidPool(cStorPoolUpdated *apis.CStorPool) error {
 	if cStorPoolUpdated.Spec.PoolSpec.PoolName == "" {
 		return fmt.Errorf("Poolname cannot be empty")
 	}
+	if len(cStorPoolUpdated.Spec.Disks.DiskList) < 1 {
+		return fmt.Errorf("Disk name(s) cannot be empty")
+	}
 	return nil
 }
 
 // GetPoolName return the pool already created.
 func GetPoolName() (string, error) {
-	poolnameStr := "zpool status | grep pool:"
+	poolnameStr := PoolOperator + " status | grep pool:"
```

Comment:
yo  can use zpool get name

Reply:
zpool get name shows output in table format, which requires even more string processing. status gives poolname in first line, so it is easy to get

Reply:
no it does not -- use zpool get -Hrp name

Reply:
ok, will change